grpc-java/grpc-servlet-jakarta/src/main/java/io/grpc/servlet/jakarta/ServletAdapter.java

```diff
@@ -172,8 +172,10 @@ public void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOEx
                 getAuthority(req),
                 logId);
 
-        transportListener.streamCreated(stream, method, headers);
-        stream.transportState().runOnTransportThread(stream.transportState()::onStreamAllocated);
+        stream.transportState().runOnTransportThread(() -> {
+            transportListener.streamCreated(stream, method, headers);
```

Comment:
i'm struggling to imagine a case where this would matter to be handled like this, but this is technically more in line with what the javadocs for ServerTransportListener specify.